src/googleAnalytics.js

```diff
@@ -0,0 +1,4 @@
+import ReactGA from 'react-ga';
+
+export const initializeGA = () => ReactGA.initialize(process.env.GA_TRACKING_ID);
+export const trackCurrentPage = () => ReactGA.pageview(window.location.pathname);
```

Comment:
Ideally this track page should be called only if GA has been initialised

Reply:
Yes I agree. I've read through react-ga docs again. There's no mention of a method that checks to see if GA is initialized. I guess instead of export const each function like I've done, I could export an object with these two functions as methods and a property called something like 'hasGABeenInit' which is a boolean and everytime the init function is called, 'hasGABeenInit' will then be set to true. Would that be the best approach?

Reply:
@nitish24p I've changed the code in `googleAnalytics.js` is this better?

```
export default {
  hasGABeenInit: false,
  initGA: () => {
    ReactGA.initialize(process.env.REACT_APP_GA_TRACKING_ID);
    this.hasGABeenInit = true;
  },
  trackCurrentPage: () => {
    if (this.hasGABeenInit) {
      ReactGA.pageview(window.location.pathname);
    }
  }
}
```

Reply:
The `this` context here will be the global window object not the object.. The idea is good but remove all the arrow functions for object method declarations. You can read more here https://dmitripavlutin.com/when-not-to-use-arrow-functions-in-javascript/
```javascript
export default {
  hasGABeenInit: false,
  initGA: function() {
    ReactGA.initialize(process.env.REACT_APP_GA_TRACKING_ID);
    this.hasGABeenInit = true;
  },
  trackCurrentPage: function() {
    if (this.hasGABeenInit) {
      ReactGA.pageview(window.location.pathname);
    }
  }
}
```